### PR TITLE
[WIP] Revolver mod location audit

### PR DESCRIPTION
**Pull request by DPavonis**
#### Summary
Balance "Revolver mod location audit"

#### Purpose of change

An audit of the game's various revolvers shows they all by default have "sights" and "stock" in their mod locations, with many also having "rail" and "underbarrel" locations as well despite all such revolvers' real-world counterparts  most decidedly _not_ supporting such modifications in their base configurations without non-trivial modification by the end-user. This PR aims to address this discrepancy. 

#### Describe the solution

~~"Borrow" ShadowFerret's much better verbiage for this kind of PR~~

Change pistol_revolver's "sights" and "stock" to their respective mounts, remove valid_mod_location definitions from all revolvers that have them so they inherit from pistol_revolver.

#### Testing

Boot up game, make sure all touched revolvers have the appropriate "[location] mount" instead of "[location]"

#### Additional context
Attached is an archive of the various revolvers demonstrating a complete lack of any sort of significantly modifiable sight, stock, rail or underbarrel location
[cata_revolvers_irl.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/9483523/cata_revolvers_irl.zip)

### CODE REVIEW

**Comment by TheShadowFerret** [comment]
>"Borrow" ShadowFerret's much better verbiage for this kind of PR

🦦

The S&W 500 could use a built_in_mod muzzle brake to both prevent suppressed revolver shenanigans and account for it having one IRL, but I don't have any other criticisms.

image

**Comment by DPavonis** [comment]
> > "Borrow" ShadowFerret's much better verbiage for this kind of PR
> 
> 🦦
> 
> The S&W 500 could use a built_in_mod muzzle brake to both prevent suppressed revolver shenanigans and account for it having one IRL, but I don't have any other criticisms.

Good point 👍

**Comment by DPavonis** [comment]
TODO: lower the loudness_modifier on both the cylinder gap and sealer after doing some research on exactly how much gas escapes on average, thus causing additional noises. Maybe add a small buff to damage to represent sealing the cylinder?

**Comment by RelMayers** [comment]
Are you aware that silencers in revolvers are:
A: lacking way to attach
B: not impactful

**Comment by RelMayers** [comment]
> TODO: lower the loudness_modifier on both the cylinder gap and sealer after doing some research on exactly how much gas escapes on average, thus causing additional noises. Maybe add a small buff to damage to represent sealing the cylinder?

I have saw only one revolver so far with seal on chamber, that is nagant.

File: data/json/items/gun/500.json [hunk 1]
```diff
@@ -60,15 +60,7 @@
...
-    "valid_mod_locations": [
-      [ "accessories", 2 ],
-      [ "grip", 1 ],
-      [ "mechanism", 4 ],
-      [ "rail", 1 ],
-      [ "sights", 1 ],
-      [ "stock mount", 1 ],
-      [ "underbarrel mount", 1 ]
-    ],
+	"build_in_mods": [ "muzzle_brake" ],
```
Comment (TheShadowFerret) [inline data/json/items/gun/500.json:63]: There should still be a muzzle slot on this, so people don't have to scroll all the way down the item description to see that it's there.

**Comment by DPavonis** [comment]
> Are you aware that silencers in almost all revolvers are: A: lacking way to attach B: not impactful

Yes yes, cylinder gap, I'm aware, and ideally any such modifications to suppress the revolver would be permanent to represent the survivor having to drill **very precise** holes in the barrel to release gas into an equally permanently-affixed suppressor

> I only know one revolver so far with seal on chamber, that is nagant.
From the **factory**, yes, the Nagant revolver is one of the few models of revolver that seals the gap between the cylinder and barrel. The cylinder sealing mod isn't intended to represent that. It's instead intended to represent something along the lines of a Peters PSDR III, which retrofits a Smith & Wesson 625 with a clamshell device over the cylinder, on top of an obturating ring to seal the cylinder gap. There are other examples out there of the engineering work that it would take to suppress a revolver, but I'm intending to implement something along the lines of a PSDR since that seems like a more plausible way for a survivor to realistically suppress a revolver than just sticking a regular old suppressor on the end of a revolver and having it work just like it would for an autoloader.



Pictured, the only halfway decent image I could find of a PSDR:

psdr-3-660x343

**Pull request closed without merging**